lib/indent-parser.ts

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash';
+
+export {
+  parse,
+};
+
+// TODO: sort out consistent interface usage
+export interface ParsedTree {
+  [dep: string]: {
+    'NUGET': object;
+  };
+}
+
+// possibly unnecessary repeated parsing without
+// taking advantage of local knowledge of paket.lock particulars
+const traverse = (node: Node, json: any) => {
+  if (node.children.length === 0) {
+    json[node.data] = null;
+  } else {
+    json[node.data] = {};
+    for (const child of node.children) {
+      traverse(child, json[node.data]);
+    }
+  }
+  return json;
+};
+
+class Node {
```

Comment:
What is a `Node`? is it a `subTree`?

Reply:
it's a node in a tree. It can be the root node (which has the reference to the whole tree structure) or a subTree node.
We are using the `Node` structure for the purpose of the parser only. This will be used to generate a depTree outside of this generic indentation-parser.